runtime/src/main/cpp/ObjCExport.mm

```diff
@@ -440,6 +440,9 @@ static OBJ_GETTER(boxedBooleanToKotlinImp, NSNumber* self, SEL cmd) {
 
 // Note: replaced by compiler in appropriate compilation modes.
 __attribute__((weak)) convertReferenceFromObjC* Kotlin_ObjCExport_blockToFunctionConverters = nullptr;
+__attribute__((weak)) int Kotlin_ObjCExport_blockToFunctionConverters_size = 0;
+
+extern "C" id objc_retainBlock(id self);
```

Comment:
Why block retention is different from objects?

Reply:
There stack-allocated blocks, which are not exactly reference-counted. `objc_retain` on these blocks is no-op.

Stack-allocated blocks can be copied to heap, a copy is reference-counted and thus "escapable". `objc_retainBlock` on stack-allocated block performs such copying, and the result has RC=1; for heap-allocated blocks it simply increments RC.